tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/generalize_zip_codes_test.py

```diff
@@ -0,0 +1,143 @@
+                (1010, 1010, 0, '2020-08-30', 1010, 'StreetAddress_PIIZIP', 1585250, '72321')
+        """).render(fq_dataset_name=self.fq_dataset_name)
+        queries.append(zipcodes_tmpl)
+
+        self.load_test_data(queries)
+
+        tables_and_counts = [{
+            'fq_table_name':
+                '.'.join([self.fq_dataset_name, 'observation']),
+            'fq_sandbox_table_name':
+                '',
+            'loaded_ids': [
+                1001, 1002, 1003, 1004, 1005, 1006, 1007, 1008, 1009, 1010
+            ],
+            'sandboxed_ids': [],
+            'fields': [
+                'observation_id', 'person_id', 'observation_concept_id',
+                'observation_date', 'observation_type_concept_id',
+                'observation_source_value', 'observation_source_concept_id',
+                'value_as_string'
+            ],
+            'cleaned_values': [
+                (1001, 1001, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1001,
+                 'StreetAddress_PIIZIP', 1585250, '605**'),
+                (1002, 1002, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1002,
+                 'StreetAddress_PIIZIP', 1585250, '108**'),
+                (1003, 1003, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1003,
+                 'StreetAddress_PIIZIP', 1585250, '972**'),
+                (1004, 1004, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1004,
+                 'StreetAddress_PIIZIP', 1585250, '934**'),
+                (1005, 1005, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1005,
+                 'StreetAddress_PIIZIP', 1585250, '005**'),
+                (1006, 1006, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1006,
+                 'StreetAddress_PIIZIP', 1585250, '235**'),
+                (1007, 1007, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1007,
+                 'StreetAddress_PIIZIP', 1585250, '235****'),
+                (1008, 1008, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1008,
+                 'StreetAddress_PIICity', 1585248, 'New York'),
+                (1009, 1009, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1009,
+                 'StreetAddress_PIIState', 1585249, 'NY'),
+                (1010, 1010, 0, datetime.strptime('2020-08-30',
+                                                  '%Y-%m-%d').date(), 1010,
+                 'StreetAddress_PIIZIP', 1585250, '723**'),
+            ]
+        }]
```

Comment:
Final addition for added record test

```suggestion
                (1010, 1010, 0, datetime.strptime('2020-08-30',
                                                  '%Y-%m-%d').date(), 1010,
                 'StreetAddress_PIIZIP', 1585250, '723**'),
                (1011, 1011, 0, datetime.strptime('2020-08-30',
                                                  '%Y-%m-%d').date(), 1011,
                 'StreetAddress_PIIZIP', 1585247, '37218')
            ]
        }]
```